### PR TITLE
Add support for geometry option to gui-metal

### PR DESCRIPTION
**Pull request by afh**
I noticed that for gut-metal `drawterm` ignores the `-g geometry` option and always places the `drawterm` window in the center of the screen at ¾ of the screen size.

The changes in this PR set the position and size of the `drawterm` window according to the `-g geometry` option and defaulting to the existing behaviour when the `-g` option has not been given.

### CODE REVIEW

**Comment by jxy** [comment]
Thanks.  I guess you want it to be the same as X11.  In this case, what would happen if I give the following: `width`, `width-off`, `widthxheight-off`, `xheight`, `xheight-off`, `-off`, `-off-off`, ...

**Comment by afh** [comment]
Great feedback @jxy, I went ahead and added [ParseGeom.c from Xenocara](https://github.com/openbsd/xenocara/blob/master/lib/libX11/src/ParseGeom.c) to handle the parsing of the geometry string.

To answer your question when given:
* `width`: `drawterm`'s width is `width` and the height is calculated from the given `width` using the golden ration (1.61833), `x` and `y` are `0`
* `width-off`: `drawterm`'s width is `width` and the height is calculated from the given `width` using the golden ration (1.61833), `x` is `off` and `y` is `0`
* `widthxheight-off`: `drawterm`'s width is `width` and the height is `height` , `x` is `off` and `y` is `0`
* `xheight`: `drawterm`'s height is `height` and the `width` is calculated from the given height using the golden ration (1.61833), `x` and `y` are `0`
* `xheight-off`: `drawterm`'s height is `height` and the `width` is calculated from the given height using the golden ration (1.61833), `x` is `off` and `y` is `0`
* `-off`: `drawterm`'s width is ¾ of the screen width and the `height` ¾ of the screen height, `x` is `off` and `y` is `0`
* `-xoff-yoff`: `drawterm`'s width is ¾ of the screen width and the `height` ¾ of the screen height, `x` is `xoff` and `y` is `off`

**Comment by afh** [comment]
@jxy Mind having another look at the latest changes to this PR?

**Comment by jxy** [comment]
Just to make sure, is the geometry passed in for the frame rectangle? Do we still need `config.h`?

**Comment by afh** [comment]
Thanks for the feedback @jxy.

I'm not sure I entirely understand your question about the frame rectangle, could you rephrase please?
To answer what I think you asked:
Yes, If drawterm is run with the `-g 1200x900+50-100` option then the resulting drawterm window on macOS will be 1200 points wide 900 points high and it is placed 50 points from the top of the screen and 100 points from the right of the screen.

The changes in this PR do not contain `config.h` file. If on the other hand you refer to the inclusion of `config.h` in [ParseGeom.c:30f](https://github.com/Plan9-Archive/drawterm-metal-cocoa/pull/5/files#diff-4e73c4a65f1c709ea6d7d75b94f5d02ffcc2446adaf458644839b64af86f5edbR30-R32), then no that isn't needed. Ideally it would be possible to include ParseGeom.c without any modifications, but if that creates problems with a `config.h` that may be generated when building drawterm, then I think it's a good idea to remove it.

**Comment by jxy** [comment]
Right, it is specified as the window size and location, but not the content size, which would be smaller.  If it's what this parameter is supposed to do, it's fine.

There's no point in leaving unused code in there, so perhaps just remove the whole ifdef.
